components/list/src/provider.rs

```diff
@@ -2,6 +2,9 @@
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
+// Provider structs must be stable
+#![allow(clippy::exhaustive_structs)]
```

Comment:
Hmm, the provider layout assumes that `ListStyle` is exhaustive. Will the compiler detect the out of bounds in e.g. line 43 or do we have to add checks there?

Reply:
It'll panic if out of bounds, though in this case the compiler will probably optimize the panic out at compile time

Reply:
Yes but now if we add a new `ListStyle` in the future, `ListFormatterPatternsV1` will become panicky.

Reply:
oh I see. We have code like

```
    pub(crate) fn start(&self, style: ListStyle) -> &ConditionalListJoinerPattern<'data> {
        &self.0[4 * (style as usize)]
    }
```

which will become panicky if `ListStyle` changes. Yeah, worth fixing. I would lean toward fixing up the provider code to not have that assumption and failing to deserialize or falling back to a different style.